### PR TITLE
Add scalar blinding for X25519

### PR DESCRIPTION
**Pull request by kaidokert**
x25519_blinded / x25519_base_blinded compute the X25519 shared secret using a per-invocation blinded scalar k' = k + r·(8·ℓ), where r is a fresh 32-bit draw from a caller-supplied CryptoRng. The output equals plain x25519 for any curve-subgroup u-coordinate.

## Summary by Sourcery

Introduce scalar-blinded X25519 operations to harden long-lived secret scalars against side-channel attacks while refactoring the Montgomery ladder for reuse with variable-length scalars.

New Features:
- Add x25519_blinded and x25519_base_blinded APIs that compute X25519 shared secrets and public keys using per-call scalar blinding driven by a caller-supplied CryptoRng.
- Expose GROUP_ORDER_BYTES constant representing the full curve group order 8·ℓ for use as the blinding modulus.

Enhancements:
- Refactor the Montgomery ladder core into a reusable montgomery_ladder function parameterized by scalar bit length to support both standard and blinded scalars.

Build:
- Add rand_core as a library dependency and rand_chacha as a dev-dependency for RNG-backed blinding and tests.

Tests:
- Add fixed-bigint-based tests verifying blinded X25519 operations match unblinded results for subgroup inputs, including edge cases such as zero blinder and different RNG seeds.

## Summary by CodeRabbit

* **New Features**
  * Added X25519 scalar blinding and new blinded key-exchange APIs; exports updated to expose blinded variants and related constants.

* **Chores**
  * Updated RNG-related dependencies (no-default-features to support no-std usage).

* **Tests**
  * Added deterministic and corpus-driven tests to validate blinded vs. unblinded X25519 behavior.

### CODE REVIEW

File: ed25519/tests/x25519_blinding.rs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        *b = ((seed.wrapping_mul(i as u64 + 1)) as u8) ^ salt ^ (i as u8);
+    }
+    out
+}
+
+#[test]
+fn blinded_matches_unblinded_x25519() {
+    // `u` must be a subgroup point for the `8·ℓ` blinding identity to
+    // hold. Random 32-byte bytes land on the twist ~half the time;
+    // derive `u` via `x25519_base` instead.
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply ed25519/tests/x25519_blinding.rs]: @gemini-code-assist Comment removed in 1816836. The choice of `x25519_base` for `u` stands on its own as a deterministic test input.

File: ed25519/src/x25519.rs [hunk 2]
```diff
@@ -48,6 +66,41 @@ pub const fn clamp(mut k: [u8; 32]) -> [u8; 32] {
...
+    }
+    for byte in out.iter_mut().skip(32) {
+        let s = (*byte as u16) + c;
+        *byte = s as u8;
+        c = s >> 8;
+    }
+    debug_assert_eq!(c, 0);
+
+    out
+}
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply ed25519/src/x25519.rs]: @gemini-code-assist Applied in fc08623 — `compute_blinded_scalar` now returns `Zeroizing<[u8; BLINDED_SCALAR_BYTES]>` directly. The wider 68-byte buffer wipes on drop within the function (panic safety) and the caller drops the wrapper at scope end.

File: ed25519/src/x25519.rs [hunk 3]
```diff
@@ -152,26 +205,168 @@ where
...
+        bits,
+        MAX_T_BYTES * 8
+    );
+
+    let p = T::from_bytes_le(&P_BYTES);
+    let field = Curve25519FieldCt::new(p).unwrap();
+
+    let k_clamped = zeroize::Zeroizing::new(clamp(*k));
+    let r = rng.next_u32();
+    let k_prime = zeroize::Zeroizing::new(compute_blinded_scalar(&k_clamped, r));
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply ed25519/src/x25519.rs]: @gemini-code-assist Applied in fc08623 — caller is now `let k_prime = compute_blinded_scalar(&k_clamped, r);`.

File: ed25519/src/x25519.rs [hunk 4]
```diff
@@ -152,26 +205,168 @@ where
...
+
+    let mut u_bytes = *u_in;
+    u_bytes[31] &= 0x7f;
+    let u = T::from_bytes_le(&u_bytes);
+    let x1 = field.reduce(&u);
+    let a24 = field.reduce(&T::from_bytes_le(&A24_BYTES));
+
+    let (x2, z2) = montgomery_ladder(&field, &x1, &a24, &*k_prime, BLINDED_BIT_COUNT);
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply ed25519/src/x25519.rs:297]: Conformance gap closed in fc08623 — switched the blinding modulus from `8·ℓ` to `lcm(curve_order, twist_order) = 8·ℓ·ℓ'` (`BLINDING_MODULUS_BYTES`, 64 bytes). The universal modulus annihilates every point on either group, so blinded matches unblinded for every 32-byte u-coordinate. The new `wycheproof_x25519_blinded.rs` runs all 518 Wycheproof X25519 vectors (twist + low-order + non-canonical included) through `x25519_blinded` and asserts blinded == unblinded — all pass. Cost is ~2× unblinded (544 vs 255 ladder iterations).

File: ed25519/tests/x25519_blinding.rs [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    }
+    impl rand_chacha::rand_core::CryptoRng for ZeroRng {}
+
+    let k = deterministic_bytes(0x42, 0xff);
+    let u = deterministic_bytes(0x42, 0x01);
+
+    let unblinded = x25519::<T>(&k, &u);
+    let blinded = x25519_blinded::<T, _>(&mut ZeroRng, &k, &u);
+
+    assert_eq!(blinded, unblinded);
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply ed25519/tests/x25519_blinding.rs:77]: Added in fc08623 as `blinded_with_max_blinder_matches_unblinded` — uses a `u32::MAX`-returning `CryptoRng` to drive full carry propagation through the 64-byte schoolbook multiply and the high end of the now-544-bit ladder.
